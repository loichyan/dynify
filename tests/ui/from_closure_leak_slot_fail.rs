fn main() {
    let mut slot_leaked = None;
    dynify::from_closure::<(), (), _>(|slot| {
        slot_leaked = Some(slot);
        unreachable!();
    });
}
