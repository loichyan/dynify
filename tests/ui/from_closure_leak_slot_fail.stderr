error[E0521]: borrowed data escapes outside of closure
 --> tests/ui/from_closure_leak_slot_fail.rs:4:9
  |
2 |     let mut slot_leaked = None;
  |         --------------- `slot_leaked` declared here, outside of the closure body
3 |     dynify::from_closure::<(), (), _>(|slot| {
  |                                        ---- `slot` is a reference that is only valid in the closure body
4 |         slot_leaked = Some(slot);
  |         ^^^^^^^^^^^^^^^^^^^^^^^^ `slot` escapes the closure body here
